src/runtime_src/core/pcie/driver/windows/include/XoclUser_INTF.h

```diff
@@ -420,27 +420,31 @@ typedef struct _XOCL_EXECPOLL_ARGS {
 } XOCL_EXECPOLL_ARGS, *PXOCL_EXECPOLL_ARGS;
 
 //
-// IOCTL_XOCL_PREAD_UNMGD
-// THE IOCTL IS NOT IMPLEMENTED
-// RETURN - STATUS_NOT_IMPLEMENTED
+// XOCL_PREAD_PWRITE_UNMGD_ARGS
+//Read IOCTL to unmanaged DDR memory
+// InputBuffer -  XOCL_PREAD_PWRITE_BO_UNMGD_ARGS  
+//OutputBuffer - (not used)
 
-#define IOCTL_XOCL_PREAD_UNMGD         CTL_CODE(FILE_DEVICE_XOCL_USER, 2105, METHOD_OUT_DIRECT, FILE_READ_DATA)
+#define IOCTL_XOCL_PREAD_UNMGD         CTL_CODE(FILE_DEVICE_XOCL_USER, 2105, METHOD_BUFFERED, FILE_READ_DATA)
 
-typedef struct _XOCL_PREAD_BO_UNMGD_ARGS {
-    ULONGLONG   Offset;     // IN: BO offset to read from 
-} XOCL_PREAD_BO_UNMGD_ARGS, *PXOCL_PREAD_BO_UNMGD_ARGS;
+typedef struct _XOCL_PREAD_PWRITE_UNMGD_ARGS {
+    uint32_t address_space; //must be 0. We must to keep it to make the structure compatible with Linux code
+    uint32_t pad;           //Currently is unused. We must to keep it to make the structure compatible with Linux code
+    uint64_t paddr;         //Physical address in the specified address space
+    uint64_t size;          //Length of data to write
+    uint64_t data_ptr;      //User's pointer(virtual address) to write the data to
+}XOCL_PREAD_PWRITE_UNMGD_ARGS, *PXOCL_PREAD_PWRITE_UNMGD_ARGS;
 
+#define DDR_BANK0_ADDRESS   0
+#define DDR_PAD0            0
```

Comment:
These two defines are confusing name wise.  Why DDR, why BANK, and PAD
Would it be better to remove them, and simply hard-code the 0 in shim.cpp?

Reply:
DDR bank and pad are unused now. I was recommended to keep them. I would like to keep them to make the structure identical to the Linux one.

Reply:
I am saying you should get rid of the #defines in this header file.   #defines with relatively generic names are bad and in this case not needed at all.

Reply:
Removed #define in the last commit.